core/src/main/scala/shapeless/generic.scala

```diff
@@ -1029,18 +1033,16 @@ class GenericMacros(val c: whitebox.Context) extends CaseClassMacros {
   def mkCoproductGeneric(tpe: Type): Tree = {
     def mkCoproductCases(tpe0: Type, index: Int): Tree = tpe0 match {
       case TypeRef(pre, sym, Nil) if sym.isModuleClass =>
-        val singleton = mkAttributedRef(pre, sym.asClass.module)
-        cq"p if p eq $singleton => $index"
-      case SingleType(pre, sym) =>
-        val singleton = mkAttributedRef(pre, sym)
-        cq"p if p eq $singleton => $index"
+        cq"p if p eq ${mkAttributedRef(pre, sym.asClass.module)} => $index"
+      case singleton: SingleType =>
+        cq"p if p eq ${mkAttributedRef(singleton)} => $index"
       case _ =>
         cq"_: $tpe0 => $index"
     }
 
     val coproduct = objectRef[Coproduct.type]
     val ctors = ctorsOf(tpe)
-    val repr = mkCoproductTypTree(ctors)
+    val repr = mkCoproductTpe(ctors)
```

Comment:
Now `mkCoproductTpe` is not used in shapeless anymore.
But it might be wise to keep it because people might be using it in their own macros.

Reply:
Ditto.